lib/cosmos/tools/cmd_tlm_server/limits_groups_background_task.rb

```diff
@@ -0,0 +1,121 @@
+    FUTURE_TIME = Time.new("3000")
+    PAST_TIME = Time.new("1900")
+
+    def initialize(initial_delay = 0)
+      super()
+      @initial_delay = Float(initial_delay)
+      @name = "Limits Groups"
+      @groups = get_limits_groups()
+      # Initialize all the group names as instance variables
+      @groups.each {|group| self.instance_variable_set("@#{group.downcase}", nil) }
+    end
+
+    # Enables and disables COSMOS limits groups. The group named after the
+    # passed group variable is automatically enabled and disabled when the
+    # yielded to block returns true and false respectively. In addition, any
+    # groups with the same name and a _PRI or _RED extension are also enabled
+    # or disabled with the base group.
+    #
+    # @param delay [Integer] The amount of time to wait after detecting the group
+    #   power on condition before actually enabling the group. A delay
+    #   allows time for telemetry to stability to avoid false positives.
+    # @param enable_code [Proc] Code to execute when the group is enabled
+    # @param disable_code [Proc] Code to execute when the group is disabled
+    def process_group(delay = 0, enable_code = nil, disable_code = nil)
+      group = calling_method.to_s[6..-1]
+
+      # Grab the instance variable based on the group name
+      var_name = "@#{group}"
+      var = self.instance_variable_get(var_name)
+      # Yield to the block to perform the telemetry check
+      if yield
+        # If the instance variable is not set or set in the past
+        # (by the disable logic) it means the group isn't enabled
+        # so store the current time to allow for an enable delay
+        if !var || var == PAST_TIME
+          self.instance_variable_set(var_name, Time.now)
+          # After setting the variable we need to get it again
+          var = self.instance_variable_get(var_name)
+        end
+        # After the requested delay after power on we enable the group
+        if Time.now > (var + delay)
+          # Reset the instance variable to a distance future time
+          # so it won't satisfy any of the other checks and enable the group
+          self.instance_variable_set(var_name, FUTURE_TIME)
+          @status = "Enabling group #{group.upcase} at #{Time.now}"
+          #Logger.info "Enabling group #{group.upcase}"
+          enable_limits_group(group)
+          # Call any additional enable code passed to the method
+          enable_code.call if enable_code
+        end
+      else
+        # If the instance variable is not set or set in the future
+        # (by the enable logic) it means the group isn't disabled
+        # Reset the instance variable to a distance past time so it
+        # won't satisfy any of the other checks and disable the group
+        if !var || var == FUTURE_TIME
+          self.instance_variable_set(var_name, PAST_TIME)
+          @status = "Disabling group #{group.upcase} at #{Time.now}"
```

Comment:
Status should more be used for checking_limits/sleeping.  Disabling group will show up in the messages.